Patch to add Solaris support.  This patch configures things like pkg5
names, SMF services, filesystem locations, etc.

This patch has not yet been submitted upstream.

diff --git a/manifests/params.pp b/manifests/params.pp
index e1ca7b3..5cd0877 100644
--- a/manifests/params.pp
+++ b/manifests/params.pp
@@ -196,6 +196,70 @@ class apache::params inherits ::apache::version {
     $error_log           = 'error_log'
     $scriptalias         = '/var/www/cgi-bin'
     $access_log_file     = 'access_log'
+  } elsif $::osfamily == 'Solaris' {
+    $user                = 'webservd'
+    $group               = 'webservd'
+    $root_group          = 'root'
+    $apache_name         = 'web/server/apache-24'
+    $service_name        = 'network/http:apache24'
+    $httpd_dir           = '/etc/apache2/2.4'
+    $server_root         = '/usr/apache2/2.4'
+    $scriptalias         = '/var/apache2/2.4/cgi-bin/'
+    $error_log           = 'error_log'
+    $access_log_file     = 'access_log'
+    $conf_dir            = $httpd_dir
+    $confd_dir           = "${httpd_dir}/conf.d"
+    $mod_dir             = "${httpd_dir}/conf.d"
+    $mod_enable_dir      = undef
+    $vhost_dir           = "${httpd_dir}/conf.d"
+    $vhost_enable_dir    = undef
+    $conf_file           = 'httpd.conf'
+    $ports_file          = "${conf_dir}/ports.conf"
+    $logroot             = '/var/apache2/2.4/logs'
+    $logroot_mode        = undef
+    $lib_path            = '/usr/apache2/2.4/libexec'
+    $mpm_module          = 'prefork'
+    $default_ssl_cert    = undef
+    $default_ssl_key     = undef
+    $ssl_certs_dir       = '/etc/certs'
+    $suphp_addhandler    = undef
+    $suphp_engine        = off
+    $suphp_configpath    = undef
+    $mod_packages        = {
+      'dbd'           => 'web/server/apache-24/module/apache-dbd',
+      'dtrace'        => 'web/server/apache-24/module/apache-dtrace',
+      'fcgid'         => 'web/server/apache-24/module/apache-fcgid',
+      'gss'           => 'web/server/apache-24/module/apache-gss',
+      'jk'            => 'web/server/apache-24/module/apache-jk',
+      'ldap'          => 'web/server/apache-24/module/apache-ldap',
+      'lua'           => 'web/server/apache-24/module/apache-lua',
+      'security'      => 'web/server/apache-24/module/apache-security',
+      'ssl-fips-140'  => 'web/server/apache-24/module/apache-ssl-fips-140',
+      'ssl'           => 'web/server/apache-24/module/apache-ssl',
+      'wsgi-26'       => 'web/server/apache-24/module/apache-wsgi-26',
+      'wsgi-27'       => 'web/server/apache-24/module/apache-wsgi-27',
+      'wsgi-34'       => 'web/server/apache-24/module/apache-wsgi-34',
+    }
+    $mod_libs             = {
+      'php5' => 'libphp5.so',
+    }
+    $conf_template          = 'apache/httpd.conf.erb'
+    $keepalive              = 'Off'
+    $keepalive_timeout      = 15
+    $max_keepalive_requests = 100
+    $fastcgi_lib_path       = '/var/apache2/2.4/htdocs/fcgid'
+    $mime_support_package   = 'system/core-os'
+    $mime_types_config      = '/etc/mime.types'
+    $docroot                = '/var/apache2/2.4/htdocs'
+    $cas_cookie_path        = undef
+    $modsec_crs_package     = undef
+    $modsec_crs_path        = undef
+    $modsec_dir             = undef
+    $modsec_default_rules   = undef
+    $error_documents_path   = '/var/apache2/2.4/error'
+    $wsgi_socket_prefix     = undef
+    $alias_icons_path       = '/var/apache2/2.4/icons'
+    $pidfile                = '/system/volatile/apache2/2.4/httpd.pid'
   } elsif $::osfamily == 'Debian' {
     $user                = 'www-data'
     $group               = 'www-data'
